mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,868 @@
+
+                    coor_x, coor_y = (
+                        x - pc_range[0]
+                    ) / voxel_size[0] / self.train_cfg['out_size_factor'], (
+                        y - pc_range[1]
+                    ) / voxel_size[1] / self.train_cfg['out_size_factor']
+
+                    ct = torch.tensor([coor_x, coor_y],
+                                      dtype=torch.float32,
+                                      device=device)
+                    ct_int = ct.to(torch.int32)
+
+                    # throw out not in range objects to avoid out of array
+                    # area when creating the heatmap
+                    if not (0 <= ct_int[0] < feature_map_size[0]
+                            and 0 <= ct_int[1] < feature_map_size[1]):
+                        continue
+
+                    draw_gaussian(heatmap[cls_id], ct_int, radius)
+
+                    new_idx = k
+                    x, y = ct_int[0], ct_int[1]
+
+                    assert (y * feature_map_size[0] + x <
+                            feature_map_size[0] * feature_map_size[1])
+
+                    ind[new_idx] = y * feature_map_size[0] + x
+                    mask[new_idx] = 1
+                    # TODO: support other outdoor dataset
+                    vx, vy = task_boxes[idx][k][6:8]
+                    rot = task_boxes[idx][k][8]
+                    if not self.train_cfg['no_log']:
+                        anno_box[new_idx] = torch.cat([
+                            ct - torch.tensor([x, y], device=device),
+                            z.unsqueeze(0), task_boxes[idx][k][3:6].log(),
+                            vx.unsqueeze(0),
+                            vy.unsqueeze(0),
+                            torch.sin(rot).unsqueeze(0),
+                            torch.cos(rot).unsqueeze(0)
+                        ])
+                    else:
+                        anno_box[new_idx] = torch.cat([
+                            ct - torch.tensor([x, y], device=device),
+                            z.unsqueeze(0), task_boxes[idx][k][3:6],
+                            vx.unsqueeze(0),
+                            vy.unsqueeze(0),
+                            torch.sin(rot).unsqueeze(0),
+                            torch.cos(rot).unsqueeze(0)
+                        ],
+                                                      dim=0)
+
+            heatmaps.append(heatmap)
+            anno_boxes.append(anno_box)
+            masks.append(mask)
+            inds.append(ind)
+        return heatmaps, anno_boxes, inds, masks
+
+    def loss(self, gt_bboxes_3d, gt_labels_3d, preds_dicts, **kwargs):
+        """Loss function for CenterHead.
+
+        Args:
+            gt_bboxes_3d (list[:obj:`LiDARInstance3DBoxes`]): Ground
+                truth gt boxes.
+            gt_labels_3d (list[torch.Tensor]): Labels of boxes.
+            preds_dicts (dict): Output of forward function.
+
+        Returns:
+            dict: Computed losses.
+
```

Comment:
Some times there may not be so many tasks, we can simplify the description here as:
```python
Returns:
    dict[str:torch.Tensor]: Loss of heatmap and bbox of each task.
```